ui/jobs/jobs.js

```diff
@@ -2491,6 +2499,105 @@ class Bars {
   }
 
   setupBrd() {
+    // DoT
+    const causticBiteBox = this.addProcBox({
+      id: 'brd-procs-causticbite',
+      fgColor: 'brd-color-causticbite',
+    });
+    const stormbBiteBox = this.addProcBox({
+      id: 'brd-procs-stormbite',
+      fgColor: 'brd-color-stormbite',
+    });
+    [
+      EffectId.Stormbite,
+      EffectId.Windbite,
+    ].forEach((effect) => {
+      this.mobGainEffectFromYouFuncMap[effect] = () => {
+        stormbBiteBox.duration = 0;
+        stormbBiteBox.duration = 30 - 0.5;
+      };
+    });
+    [
+      EffectId.CausticBite,
+      EffectId.VenomousBite,
+    ].forEach((effect) => {
+      this.mobGainEffectFromYouFuncMap[effect] = () => {
+        causticBiteBox.duration = 0;
+        causticBiteBox.duration = 30 - 0.5;
+      };
+    });
+    this.statChangeFuncMap['BRD'] = () => {
+      stormbBiteBox.valuescale = this.gcdSkill();
+      stormbBiteBox.threshold = this.gcdSkill() * 2;
+      causticBiteBox.valuescale = this.gcdSkill();
+      causticBiteBox.threshold = this.gcdSkill() * 2;
+      songBox.valuescale = this.gcdSkill();
+    };
+
+    // Song
+    const songBox = this.addProcBox({
+      id: 'brd-procs-song',
+      fgColor: 'brd-color-song',
+    });
+    const repertoireBox = this.addResourceBox({
+      classList: ['brd-color-song'],
+    });
+    this.repertoireTimer = this.addTimerBar({
+      id: 'brd-timers-repertoire',
+      fgColor: 'brd-color-song',
+    });
+    // Only with-DoT-target you last attacked will trigger this timer.
+    // So it work not well in mutiple targets fight.
+    this.UpdateDotTimer = () => {
+      this.repertoireTimer.duration = 2.91666;
```

Comment:
Ohmygosh, this is GREAT.  I haven't played bard in ages, but having a timer for when your procs might come up just feels really good.

Reply:
Is the internal cooldown on Repertoire not exactly 3 seconds? Or is this value here intended to adjust for some sort of interface delay?

Reply:
Yes, the Reperoire tick (or server DoT tick) is in every "Eorzea minute".
We know that 1 "Eorzea day" == 70 "Earth minutes", so 1 "Eorzea minute" equals `(70/(1*24*60))*60 = `2.91667 "Earth seconds".

Reply:
Ahh, okay. I had always internalized it as 3s and never bothered to time it exactly after it became muscle memory. Thanks for confirming that!